frontend/src/components/project-menu/ProjectMenuCard.tsx

```diff
@@ -58,20 +60,23 @@ Finally, open up a pull request using the GitHub API and the token in the GITHUB
     setContextMenuIsOpen(false);
   };
 
+  const handleDownloadWorkspace = () => {
+    posthog.capture("download_workspace_button_clicked");
+    try {
+      downloadWorkspace();
+    } catch (error) {
+      toast.error("Failed to download workspace");
```

Comment:
While we're in here, how do we feel about sending this GitHub connection error as an event as well?

Reply:
I would agree, there are a lot of places we can catch errors like this. After being reminded about the free tier though, maybe we can hold off on this until 1-2 weeks from now. Just to give us time to see how it fares. Let me know if you or @rbren think otherwise and I'll do it in a follow-up PR